src/tracks/playabletrack/notetrack/ui/NoteTrackAffordanceControls.cpp

```diff
@@ -0,0 +1,83 @@
+#include "NoteTrackAffordanceControls.h"
+
+#include "../../../ui/AffordanceHandle.h"
+#include "../../../../AllThemeResources.h"
+#include "../../../../AColor.h"
+#include "../../../../NoteTrack.h"
+#include "../../../../ViewInfo.h"
+#include "../../../../TrackArtist.h"
+#include "../../../../TrackPanelMouseEvent.h"
+#include "../../../../TrackPanel.h"
+#include "../../../../TrackPanelDrawingContext.h"
+
+#include "../lib-src/header-substitutes/allegro.h"
+
+
+NoteTrackAffordanceControls::NoteTrackAffordanceControls(const std::shared_ptr<Track>& pTrack)
+    : CommonTrackCell(pTrack)
+{
+
+}
+
+std::vector<UIHandlePtr> NoteTrackAffordanceControls::HitTest(const TrackPanelMouseState& state, const AudacityProject* pProject)
+{
+    std::vector<UIHandlePtr> results;
+
+    auto track = FindTrack();
+    const auto nt = std::static_pointer_cast<const NoteTrack>(track->SubstitutePendingChangedTrack());
+
+    const auto rect = state.rect;
+
+    auto& trackPanel = TrackPanel::Get(*pProject);
+    auto& zoomInfo = *trackPanel.GetViewInfo();
```

Comment:
Just ViewInfo::Get(*pProject) and you can avoid use of TrackPanel

Reply:
Thanks!